tests/test_line_profiler.py

```diff
@@ -70,30 +70,18 @@ def test_function_decorator(self):
         self.assertEqual(profile.enable_count, 0)
         self.assertEqual(value, f(10))
 
-        timings = profile.code_map[f.__code__]
```

Comment:
Can we make a property that mimics this for compatibility? (even if it is slow and add a warning?)

Reply:
I believe the only way to do that would be to go back to using a python dict to store code objects. If there's a way to pass an argument to the decorator to allow for this old behavior for compatability reasons, then that could work, but it'd essentially entail having the new code and the old code side-by-side.

Reply:
Do we have access to those code objects? Is it significant overhead to also maintain that python dict with the alternate view? My main concern is because it is a Python API compatibility break. I'm find breaking the C ABI. I don't think many people are linking to this. But I think this might break people's code if its removed (which I'm more ok with if its necessary to obtain the speed improvement)

Reply:
We do have access to those code objects. However, in order to associate the python code objects with the c hash, I believe we'd need to maintain a python dict with the code objects in the trace callback, and removing that was precisely how I obtained the speedup.